src/transformers/models/wav2vec2/feature_extraction_wav2vec2.py

```diff
@@ -79,13 +79,20 @@ def __init__(
         self.do_normalize = do_normalize
 
     @staticmethod
-    def zero_mean_unit_var_norm(input_values: List[np.ndarray], input_lengths: List[int]) -> List[np.ndarray]:
+    def zero_mean_unit_var_norm(
+        input_values: List[np.ndarray], attention_mask: List[np.ndarray], padding_value=0.0
+    ) -> List[np.ndarray]:
         """
         Every array in the list is normalized to have zero mean and unit variance
         """
-        normed_input_values = [
-            (x - np.mean(x[:i])) / np.sqrt(np.var(x[:i]) + 1e-5) for x, i in zip(input_values, input_lengths)
-        ]
+
+        if attention_mask is not None:
+            normed_input_values = [
+                (x - x[:i].mean()) / np.sqrt(x[:i].var() + 1e-7) for x, i in zip(input_values, attention_mask.sum(-1))
+            ]
+        else:
+            normed_input_values = [(x - x.mean()) / np.sqrt(x.var() + 1e-7) for x in input_values]
```

Comment:
Nice!
This will also affect `Speech2TextFeatureExtractor`, no ? It also does normalization 

https://github.com/huggingface/transformers/blob/09549aa18c7c1b56106790a04c06566846de209b/src/transformers/models/speech_to_text/feature_extraction_speech_to_text.py#L95-L108

Reply:
Actually yeah that's a good point - will check!

Reply:
Ok there were a couple of problems here actually :-/ Think most of them are fixed now